apple/RNGestureHandlerModule.mm

```diff
@@ -140,13 +144,43 @@ - (void)setBridge:(RCTBridge *)bridge
 #endif // RCT_NEW_ARCH_ENABLED
 
 #ifdef RCT_NEW_ARCH_ENABLED
-RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(install)
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(installUIRuntimeBindings)
 {
+  auto runtime = _runtime;
+
   dispatch_block_t block = ^{
-    RCTCxxBridge *cxxBridge = (RCTCxxBridge *)self.bridge;
-    auto runtime = (jsi::Runtime *)cxxBridge.runtime;
-    decorateRuntime(*runtime);
+    auto arrayBufferValue = runtime->global()
+                                .getProperty(*runtime, "_WORKLET_RUNTIME")
+                                .getObject(*runtime)
+                                .getArrayBuffer(*runtime)
+                                .data(*runtime);
+    uintptr_t *uiAddr = reinterpret_cast<uintptr_t *>(&arrayBufferValue[0]);
+    jsi::Runtime *uiRuntime = reinterpret_cast<jsi::Runtime *>(*uiAddr);
+
+    __weak RNGestureHandlerModule *weakSelf = self;
+
+    auto setGestureStateNew = jsi::Function::createFromHostFunction(
+        *uiRuntime,
+        jsi::PropNameID::forAscii(*uiRuntime, "_setGestureStateNew"),
+        2,
+        [weakSelf](jsi::Runtime &runtime, const jsi::Value &thisValue, const jsi::Value *arguments, size_t count)
+            -> jsi::Value {
+          if (count == 2) {
+            auto handlerTag = (int)arguments[0].asNumber();
+            auto state = (int)arguments[1].asNumber();
+
+            // TODO: expose to JS and dispatch to UI thread if called on JS?
+            RNGestureHandlerModule *strongSelf = weakSelf;
+            if (strongSelf != nullptr) {
+              [strongSelf setGestureState:state forHandler:handlerTag];
+            }
+          }
+          return jsi::Value::undefined();
+        });
+
+    uiRuntime->global().setProperty(*uiRuntime, "_setGestureStateNew", std::move(setGestureStateNew));
   };
+
   if (_dispatchToJSThread) {
```

Comment:
Do we still need this? I think we're safe to assume that `RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(installUIRuntimeBindings)` will always be called on the JS queue.

Reply:
That sounds like a safe assumption, especially given that Android didn't have a counterpart of that.